engine/src/main/scala/cromwell/engine/workflow/WorkflowProcessingEventPublishing.scala

```diff
@@ -36,4 +38,21 @@ object WorkflowProcessingEventPublishing {
 
     serviceRegistry ! PutMetadataAction(metadata)
   }
+
+  def publishLabelsToMetadata(workflowId: WorkflowId,
+                              labels: Map[String, String],
+                              serviceRegistry: ActorRef): IOChecked[Unit] = {
+    val defaultLabel = "cromwell-workflow-id" -> s"cromwell-$workflowId"
+    Monad[IOChecked].pure(labelsToMetadata(workflowId, labels + defaultLabel, serviceRegistry))
```

Comment:
Note: this change was triggered by IntelliJ warning about a dupe block of code. That and your comment also helped me notice a bug where one of these `IOChecked[Unit]`s were being created but never run. 👍

Happy to chat in-person if I'm off target here. The `SomeBox[Unit]` vs. `Unit` is because the latter only allows throwing execptions, not returning them. Could have been as simple as a [`Try[Unit]`](https://alvinalexander.com/scala/functional-error-handling-exceptions-in-scala), but in this case the previously-duplicated-and-now-consolidated block was written to use the feature rich `IOChecked[Unit]`.

Reply:
But I don't get why this would ever throw an exception? It seems like you're assuming it won't either because later on you use `.get` in the production `WorkflowStoreSubmit` actor from the return code?

Reply:
Yeh, I didn't write it. Just de-duped it. I concur the function probably won't throw an exception. I'm pretty sure one-of-the-original-dupes was originally wrapped so it would fit into [this `IOChecked[_]` for-comprehension](https://github.com/broadinstitute/cromwell/blob/44/engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala#L270).